plugin/metrics/prometheus/metricsstore/reader.go

```diff
@@ -287,17 +288,19 @@ func promqlDurationString(d *time.Duration) string {
 	return string(b)
 }
 
-func startSpanForQuery(ctx context.Context, metricName, query string) (opentracing.Span, context.Context) {
-	span, ctx := opentracing.StartSpanFromContext(ctx, metricName)
-	ottag.DBStatement.Set(span, query)
-	ottag.DBType.Set(span, "prometheus")
-	ottag.Component.Set(span, "promql")
-	return span, ctx
+func startSpanForQuery(ctx context.Context, metricName, query string) (context.Context, trace.Span) {
+	tp := otel.GetTracerProvider()
+	ctx, span := tp.Tracer("prom-metrics-reader").Start(ctx, metricName)
```

Comment:
This is almost correct, but you don't want to be asking for the tracer on critical path of every request, since the tracer never changes. Instead, you can create the tracer in `NewMetricsReader` and store it as member variable.

Reply:
even cleaner is to make the Factory call `GetTracerProvider()` and pass the provider to `NewMetricsReader`